src/network.zig

```diff
@@ -1048,78 +1031,56 @@ pub const Protocols = struct {
-			defer main.stackAllocator.free(headeredData);
-			headeredData[0] = header;
-			@memcpy(headeredData[1..], data);
-			conn.sendImportant(id, headeredData);
-		}
-
-		fn addHeaderAndSendUnimportant(conn: *Connection, header: u8, data: []const u8) void {
-			const headeredData = main.stackAllocator.alloc(u8, data.len + 1);
-			defer main.stackAllocator.free(headeredData);
-			headeredData[0] = header;
-			@memcpy(headeredData[1..], data);
-			conn.sendUnimportant(id, headeredData);
-		}
-
 		pub fn sendGamemode(conn: *Connection, gamemode: main.game.Gamemode) void {
-			conn.sendImportant(id, &.{type_gamemode, @intFromEnum(gamemode)});
+			conn.sendImportant(id, &.{@intFromEnum(UpdateType.gamemode), @intFromEnum(gamemode)});
 		}
 
 		pub fn sendTPCoordinates(conn: *Connection, pos: Vec3d) void {
 			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, networkEndian, 25);
 			defer writer.deinit();
-			writer.writeInt(u8, type_teleport);
-			writer.writeInt(u64, @bitCast(pos[0]));
-			writer.writeInt(u64, @bitCast(pos[1]));
-			writer.writeInt(u64, @bitCast(pos[2]));
-			conn.sendImportant(id, writer.data.items);
-		}
 
-		pub fn sendCure(conn: *Connection) void {
-			var data: [1]u8 = undefined;
-			data[0] = type_cure;
-			conn.sendImportant(id, &data);
+			writer.writeEnum(UpdateType, .teleport);
+			writer.writeVec(Vec3d, pos);
+
+			conn.sendImportant(id, writer.data.items);
 		}
 
 		pub fn sendWorldEditPos(conn: *Connection, posType: WorldEditPosition, maybePos: ?Vec3i) void {
 			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, networkEndian, 25);
 			defer writer.deinit();
-			writer.writeInt(u8, type_worldEditPos);
+
+			writer.writeEnum(UpdateType, .worldEditPos);
 			writer.writeEnum(WorldEditPosition, posType);
 			if(maybePos) |pos| {
-				writer.writeInt(i32, pos[0]);
-				writer.writeInt(i32, pos[1]);
-				writer.writeInt(i32, pos[2]);
+				writer.writeVec(Vec3i, pos);
 			}
+
 			conn.sendImportant(id, writer.data.items);
 		}
 
 		pub fn sendTimeAndBiome(conn: *Connection, world: *const main.server.ServerWorld) void {
-			const zon = ZonElement.initObject(main.stackAllocator);
-			defer zon.deinit(main.stackAllocator);
-			zon.put("time", world.gameTime);
-			const pos = conn.user.?.player.pos;
-			zon.put("biome", (world.getBiome(@intFromFloat(pos[0]), @intFromFloat(pos[1]), @intFromFloat(pos[2]))).id);
-			const string = zon.toString(main.stackAllocator);
-			defer main.stackAllocator.free(string);
-			addHeaderAndSendUnimportant(conn, type_timeAndBiome, string);
+			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, networkEndian, 25);
+			defer writer.deinit();
+
+			writer.writeEnum(UpdateType, .timeAndBiome);
+			writer.writeInt(i64, world.gameTime);
+
+			const pos = @as(Vec3i, @intFromFloat(conn.user.?.player.pos));
+			const biomeId = world.getBiome(pos[0], pos[1], pos[2]).paletteId;
+			writer.writeInt(u32, biomeId);
```

Comment:
Let's just avoid the naming problem here maybe
```suggestion
			writer.writeInt(u32, world.getBiome(pos[0], pos[1], pos[2]).paletteId);
```